deepchecks/vision/checks/distribution/train_test_label_drift.py

```diff
@@ -154,57 +134,24 @@ def initialize_run(self, context: Context):
-                           zip(self._bounds_list, self._num_bins_list)]
-        self._train_hists = [x[0] for x in hists_and_edges]
-        self._test_hists = [copy(hist) for hist in self._train_hists]
-        self._edges = [x[1] for x in hists_and_edges]
-
-        self._train_counters = [Counter() for i in range(num_discrete_transformers)]
-        self._test_counters = [Counter() for i in range(num_discrete_transformers)]
+        self._train_label_properties = OrderedDict([(k['name'], []) for k in self._label_measurements])
+        self._test_label_properties = OrderedDict([(k['name'], []) for k in self._label_measurements])
 
     def update(self, context: Context, batch: Any, dataset_kind):
         """Perform update on batch for train or test counters and histograms."""
         # For all transformers, calculate histograms by batch:
         if dataset_kind == DatasetKind.TRAIN:
-            train_dataset = context.train
-            self._train_hists = calculate_continuous_histograms_in_batch(batch, self._train_hists,
-                                                                         self._continuous_label_measurements,
-                                                                         self._bounds_list, self._num_bins_list,
-                                                                         train_dataset)
-            self._train_counters = calculate_discrete_histograms_in_batch(batch, self._train_counters,
-                                                                          self._discrete_label_measurements,
-                                                                          train_dataset)
-
+            dataset = context.train
+            properties = self._train_label_properties
         elif dataset_kind == DatasetKind.TEST:
-            test_dataset = context.test
-            self._test_hists = calculate_continuous_histograms_in_batch(batch, self._test_hists,
-                                                                        self._continuous_label_measurements,
-                                                                        self._bounds_list, self._num_bins_list,
-                                                                        test_dataset)
-            self._test_counters = calculate_discrete_histograms_in_batch(batch, self._test_counters,
-                                                                         self._discrete_label_measurements,
-                                                                         test_dataset)
+            dataset = context.test
+            properties = self._test_label_properties
         else:
             raise DeepchecksNotSupportedError(f'Unsupported dataset kind {dataset_kind}')
 
+        for label_measurement in self._label_measurements:
+            properties[label_measurement['name']] += get_results_on_batch(batch, label_measurement['method'], dataset)
```

Comment:
why not using extend? you will have in the end just a single list instead of list of lists